llama/sap.py

```diff
@@ -6,6 +6,7 @@
 from datetime import datetime
 from math import fsum
 from typing import List, Literal, Optional, Tuple
+import flatdict
```

Comment:
I don't _love_ importing a third-party library to do just one small thing, but I do see the value in having the nested keys to help identify exactly which field the multibyte character is in. So I think we can go with it.

Reply:
Agree. I wanted to be able to show the nested keys, which is exactly what this library does. It was going to take me a long time to figure out how to do that on my own.